com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -475,9 +476,13 @@ public virtual void InitializeAgent()
         /// </returns>
         public virtual float[] Heuristic()
         {
-            throw new UnityAgentsException(
-                "The Heuristic method was not implemented for the Agent on the " +
-                $"{gameObject.name} GameObject.");
+            Debug.LogWarning("Heuristic method called but not implemented. Return placeholder actions.");
+            var param = m_PolicyFactory.brainParameters;
+            var actionSize = param.vectorActionSpaceType == SpaceType.Continuous ?
```

Comment:
Feels like this should be a property on brainParameters since we do it in ResetData() too :
https://github.com/Unity-Technologies/ml-agents/blob/78a218b4254a74c5580aa3451ee9e391d65122ff/com.unity.ml-agents/Runtime/Agent.cs#L443-L452

Reply:
`int BrainParameters.ActionAllocationSize()`?

Reply:
IMO, vectorActionSize should be an int and in discrete control only there should be an array of ints called discreteActionBrancheSizes.

Reply:
That sounds like a bigger change that's out of scope for this PR. Are you OK leaving this code as is, or do you think a utility method/property would be cleaner?